networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/DiscoveryPeer.java

```diff
@@ -14,21 +14,28 @@
 package tech.pegasys.teku.networking.p2p.discovery;
 
 import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
 import java.net.InetSocketAddress;
-import java.util.Objects;
+import java.util.List;
 import java.util.Optional;
 import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.teku.datastructures.networking.libp2p.rpc.EnrForkId;
 
 public class DiscoveryPeer {
   private final Bytes publicKey;
   private final InetSocketAddress nodeAddress;
-  private final Optional<Bytes> enrForkId;
+  private final Optional<EnrForkId> enrForkId;
+  private final Optional<List<Integer>> persistentSubnets;
 
   public DiscoveryPeer(
-      final Bytes publicKey, final InetSocketAddress nodeAddress, final Optional<Bytes> enrForkId) {
+      final Bytes publicKey,
+      final InetSocketAddress nodeAddress,
+      final Optional<EnrForkId> enrForkId,
+      final Optional<List<Integer>> persistentSubnets) {
```

Comment:
I think it's probably better to just use an empty list if the attribute is not present rather than having an `Optional<List>`.  Not advertising any persistent subnets is effectively the same as not having any.

Also it would probably be better to stick with the `Bitvector` rather than converting to `List<Integer>` the most common thing we'll do is check for a specific value which in a `List` means iterating through every item but is as single lookup in a `Bitvector`.

Reply:
Makes sense to me.